File: sudo.py
import os, sys

# Requires Administrator (sudo)
if os.geteuid() != 0:
    print("Program requires Administrator privileges. Restarting...")
    os.execvp("sudo", ["sudo", sys.executable] + sys.argv)

# Python
os.chdir(os.path.dirname(os.path.abspath(__file__)))

# Run Program main
print("Welcome, I'm Mohammed Al-Baqer")

# Simulate program execution
try:
    pass
except Exception as e:
    print(f"Failed to execute the program: {e}")
    input("Press Enter to exit...")

File: sudo.cpp
#include <iostream>
#include <cstdlib>
#include <unistd.h>

void RunAsAdministrator() {
    std::cout << "Program requires Administrator privileges. Restarting...\n";
    execlp("sudo", "sudo", "/proc/self/exe", NULL);
}

int main() {
    if (geteuid() != 0) {
        RunAsAdministrator();
        return 0;
    }

    std::cout << "Welcome, I'm Mohammed Al-Baqer\n";

    try {
    } catch (const std::exception& e) {
        std::cerr << "Failed to execute the program: " << e.what() << std::endl;
        system("read -p 'Press Enter to exit...' var");
    }

    return 0;
}
